packages/typespec-vscode/src/emit/emitter.ts

```diff
@@ -0,0 +1,42 @@
+import vscode from "vscode";
+import { SettingName } from "../types.js";
+
+export enum EmitterKind {
+  Schema = "schema",
+  Client = "client",
+  Server = "server",
+}
+
+export interface Emitter {
+  language: string;
+  package: string;
+  version?: string;
+  kind: EmitterKind;
+}
+
+const extensionConfig = vscode.workspace.getConfiguration();
+
+function getEmitter(kind: EmitterKind, emitter: Emitter): Emitter {
+  const packageFullName: string = emitter.package ?? "";
+  const index = packageFullName.lastIndexOf("@");
+  let version = undefined;
+  let packageName = packageFullName;
+  if (index !== -1 && index !== 0) {
```

Comment:
trim first?

Reply:
trim will introduce a new string. Since we can filter out the invalid scenario ( @ is the first character), we may not waste another space.

Reply:
user can configure a string with leading space, can't they?

Reply:
add trim()